### PR TITLE
CDAP-18109 update connector doc and extract the ConfigUtil.java from google-cloud repo

### PR DESCRIPTION
**Pull request by seanzhougoogle**

### CODE REVIEW

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utilities for plugin configuration
+ */
+public class ConfigUtil {
```
Comment (albertshau) [inline hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: why is this needed here?

Comment (avirmani) [inline hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: where is it being used?

**Review by avirmani — APPROVED** [review]

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java (hunk 1 above)
Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: it will be used in `google-cloud` repo and `amazon-s3-plugin` repo. I will send a PR later soon.

Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: This util was previously used in `google-cloud` repo, but I found it's not `google-cloud` specific and can be exact to even higher level common layer including the new `validateConnection` method I added.

Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: see: https://github.com/data-integrations/google-cloud/pull/702

Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: see: data-integrations/google-cloud#702 and data-integrations/amazon-s3-plugins#88

File: core-plugins/src/main/java/io/cdap/plugin/batch/connector/FileConnector.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
  */
 @Plugin(type = Connector.PLUGIN_TYPE)
 @Name(FileConnector.NAME)
-@Description("File connector for local file system")
+@Description("Connection to browse and sample data from the local file system.")
```
Comment (yaojiefeng) [inline core-plugins/src/main/java/io/cdap/plugin/batch/connector/FileConnector.java:53]: Connection -> Connector

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  public static final String NAME_CONNECTION = "connection";
+  public static final String NAME_USE_CONNECTION = "useConnection";
```
Comment (yaojiefeng) [inline hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:30]: Let's put these in `Constants`

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  public static Map<String, String> parseKeyValueConfig(String configValue, String delimiter,
```
Comment (yaojiefeng) [inline hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:32]: there is a `KeyValueListParser` which does the same thing, this can be removed

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      String[] parts = property.split(keyValueDelimiter);
+      String key = parts[0];
+      String value = parts[1];
+      map.put(key, value);
+    }
+    return map;
+  }
+
+  public static String getKVPair(String key, String value, String keyValueDelimiter) {
+    return String.format("%s%s%s", key, keyValueDelimiter, value);
```
Comment (yaojiefeng) [inline hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:45]: this shouldn't need to be a helper method

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utilities for plugin configuration
+ */
+public class ConfigUtil {
```
Comment (yaojiefeng) [inline hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: Let's call this `ConnectionUtil` to be more specific, `ConfigUtil` is too general

File: core-plugins/src/main/java/io/cdap/plugin/batch/connector/FileConnector.java (hunk 2 above)
Comment (seanzhougoogle) [reply core-plugins/src/main/java/io/cdap/plugin/batch/connector/FileConnector.java:53]: this is per request of our document writer : https://cdap.atlassian.net/browse/CDAP-18109

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java (hunk 3 above)
Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:30]: Since it's already referenced in merged PR https://github.com/data-integrations/google-cloud/pull/702, I will create another PR to refactor later.

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java (hunk 4 above)
Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:32]: this class was extracted from a `google-cloud`, and is referenced in https://github.com/data-integrations/google-cloud/pull/702. I will create another PR to do the refactor.

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java (hunk 5 above)
Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:45]: this class was extracted from a google-cloud, and is referenced in data-integrations/google-cloud#702. I will create another PR to do the refactor.

File: hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java (hunk 6 above)
Comment (seanzhougoogle) [reply hydrator-common/src/main/java/io/cdap/plugin/common/ConfigUtil.java:28]: It is referenced in data-integrations/google-cloud#702. I will create another PR to do the refactor.
